chatClient.links

```diff
@@ -152,35 +152,47 @@ module Components {
   }
 
   fun chooseRoom(ap) {
-    fun makeRadioRec(rooms, handler) {
-      switch(rooms) {
-        case [] -> <#/>
-        case x::xs ->
-          var Room(roomStr) = x;
-          <#><input type="radio" name="room"
-                    l:onchange="{handler ! Room(roomStr)}"/>
-          {stringToXml(roomStr)}<br/>
-          {makeRadioRec(xs, handler)}</#>
-      }
+    fun makeRoomForm(rooms, handlerFn) {
+      fun makeRadioRec(rooms) {
+            switch(rooms) {
+              case [] -> <#/>
+              case x::xs ->
+                var Room(roomStr) = x;
+                <#>
+                  <input type="radio" name="room"
+                    l:onchange="{ handlerFn ! Room(roomStr)}"/>
+                { stringToXml(roomStr) } <br/>
+                { makeRadioRec(xs) } </#>
+            }
+          }
+      var radios = makeRadioRec(rooms);
```

Comment:
This never seems to be used later (since you are calling the "makeRadioRec" function directly from within the Xml).